tests/typecheck/managers/querysets/test_from_queryset.yml

```diff
@@ -43,6 +43,31 @@
                 class MyModelWithoutSelf(models.Model):
                     objects = ManagerWithoutSelf()
 
+-   case: from_queryset_model_inheritance
+    main: |
+        from myapp.models import Base, Sub
+        reveal_type(Base.objects)  # N: Revealed type is "myapp.models.MyManager[myapp.models.Base]"
+        reveal_type(Sub.objects)  # N: Revealed type is "myapp.models.MyManager[myapp.models.Sub]"
+    installed_apps:
+        - myapp
+    files:
+        -   path: myapp/__init__.py
+        -   path: myapp/models.py
+            content: |
+                from typing import ClassVar
+                from typing_extensions import Self
+                from django.db.models import Model
+                from django.db.models.manager import BaseManager
+                from django.db.models.query import QuerySet
+
+                MyManager = BaseManager.from_queryset(QuerySet, "MyManager")
+
+                class Base(Model):
+                    objects: ClassVar[MyManager[Self]] = MyManager()
```

Comment:
Do you think we also need to check that the plugin is able to infer the manager type in a scenario like this?

e.g. also test

```python
class Base(Model):
    objects = MyManager()
```

I suppose there might be a bunch of other tests doing that, so it might just be a duplication

Reply:
most of the tests do that already fortunately!

Reply:
Alright, lets merge